### PR TITLE
Runtime hooks: fix "spawn" multiprocessing on mac

### LINKED ISSUE #2322: multiprocessing spawn in python 3 does not work on macOS
I'm not able to get multiprocessing to work in spawn mode on macOS  when using pyinstaller. The following script runs fine in the interpreter, but when using pyinstaller it enters an infinite recursive loop that keeps starting new processes.

I'm using Python 3 from Anaconda (Python 3.5.2 |Continuum Analytics, Inc.| (default, Jul  2 2016, 17:52:12) ), on macOS 10.12.2, with the development version of pyinstaller (3.3.dev0+483c819).

```python
import sys
from multiprocessing import freeze_support, set_start_method, Process

def test():
    print('In subprocess')

if __name__ == '__main__':
    print(sys.argv,)
    freeze_support()
    set_start_method('spawn')

    print('In main')
    proc = Process(target=test)
    proc.start()
    proc.join()
```

### PR DESCRIPTION
**Pull request by xoviat**
A few notes: 

- the evidence that this works is [here](https://travis-ci.org/xoviat/mac_scratchpad/builds/210814888)
- this may break compatibility with the 'fork' method on mac; needs further investigation
- closes #2322.

### CODE REVIEW

**Comment by xoviat** [comment]
Looks like we have a segfault on 2.7/linux? Not sure what that is about or how this change could have caused it.

File: PyInstaller/loader/rthooks.dat [hunk 1]
```diff
@@ -24,4 +24,5 @@
     'usb':        ['pyi_rth_usb.py'],
     'win32com':   ['pyi_rth_win32comgenpy.py'],
     'cryptography': ['pyi_rth_cryptography.py'],
+    'multiprocessing': ['pyi_rth_multiprocessing.py'],
```
Comment (htgoebel) [inline PyInstaller/loader/rthooks.dat]: What do you think about moving this data-struct into `PyInstaller.loader__init__.py` and add this entry only on darwin?

Comment (xoviat) [inline PyInstaller/loader/rthooks.dat]: Ideally the struct would be calculated based on filenames similar to the standard hook filenames.

**Comment by xoviat** [comment]
So, not working after all.

**Comment by xoviat** [comment]
Fixed.

**Comment by htgoebel** [comment]
I was about to merge this, but this pull-request contains a lot of unrelated changes. Would you mind updating it to only include the relevant ones? Thanks.

**Comment by xoviat** [comment]
Once again, the changes should disappear after you merge the osx pull request.

**Comment by xoviat** [comment]
Should be OK based on user feedback.

**Comment by htgoebel** [comment]
Merged manually after some cleanup, see 41c426f6.

**Comment by springermac** [comment]
This has [broken](https://travis-ci.org/pyinstaller/pyinstaller/jobs/263153179#L3412) multiprocess forking on macOS.

**Comment by xoviat** [comment]
@springermac I just used @bpteague's code which osx commenters said was okay in the relevant issue. Are you planning to submit a patch or should we revert these changes (breaking "spawn" again)?

**Comment by bpteague** [comment]
I think we've fixed this issue in PR #2795.

**Pull request closed without merging**